### ISSUE TITLE
CocoaPods, Reachability, and use_frameworks!

### ISSUE THREAD

**Issue by g0ld2k** [tonymillion/Reachability#95]
With CocoaPods 0.36, they added the capability to generate frameworks for pods.  For Swift projects, using frameworks eliminates the need for a Objective-C bridge. 

Podfile:

        # Uncomment this line to define a global platform for your project
        platform :ios, "8.0"
        use_frameworks!

        target "VB ParkFinder" do
        pod 'ArcGIS-Runtime-SDK-iOS', '~> 10.2'
        pod 'Reachability', '~> 3.2'
        pod 'TUSafariActivity', '~> 1.0.3'
        pod 'NHCalendarActivity', '~> 0.0.1'
        end

        target "VB ParkFinderTests" do

        end

Reachability works great as a Framework, however when submitting the app to Apple, Apple incorrectly identifies Reachability as being a private API (it looks like there is a private API class called Reachability), which causes the app to be rejected.

>Dear developer,
>
>We have discovered one or more issues with your recent delivery for "VB ParkFinder". To process your delivery, the following issues must be corrected:
>
>Non-public API usage:
>
>The app contains or inherits from non-public classes in VB ParkFinder: Reachability
>
>If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed.
>
>If you think this message was sent in error and that you have only used Apple-published APIs in accordance with the guidelines, send the app's nine-digit Apple ID, along with detailed information about why you believe the above APIs were incorrectly flagged, to appreview@apple.com. For further information, visit the Technical Support Information page.
>
>Once these issues have been corrected, you can then redeliver the corrected binary.
>
>Regards,
>
>The App Store team

None of the other Pods in my project caused this issue.

I was able to work around the issue by removing the `use_frameworks!` line from my Podfile and creating a Objective-C bridging file, no other code was changed.  My submission without `use_frameworks!` passed Apple's initial checks without a problem.

I'm not sure what the solution would be, but it sounds like there is a collision occurring with Apple's private frameworks due to the naming of your project.

**Comment by tonymillion** [comment]
Thats a new one! The solution, I suppose, would be to rename the framework as something else while keeping the name for compiled projects?

**Comment by albertbori** [comment]
+1

Got rejected for the same reason. I'd rather not remove `use_frameworks!`. Is there another quick solution for this?

**Comment by Cinamonas** [comment]
+1

Our's got rejected due to this, too.

**Comment by formvoltron** [comment]
time for a rename it seems.

**Comment by tonymillion** [comment]
I agree - I have a lot of work right now, but I'd like to look into if you can detect when being compiled as a framework and change the class name dynamically.

**Comment by formvoltron** [comment]
Why not just TMReachability?  We'll know it's still Reachability under the
hood. :)

**Comment by salabaha** [comment]
Faced with the same problem today. Renaming should fix it.

**Comment by allaire** [comment]
:+1:

**Comment by cpatterson** [comment]
I'm also facing this problem. Can't remove `use_frameworks!` because we're using some Swift pods. 
None of the forks I've looked at seem to rename everything properly for CocoaPods.

**Comment by albertbori** [comment]
@cpatterson I forked and use this regularly to get my app into the app-store. It might be a few fixes behind. I was planning on bringing it up-to-date soon (if needed): https://github.com/albertbori/Reachability

@tonymillion I recommend renaming the items in question to `TMReachability` instead of chasing the golden helicopter solution. Especially since this issue is a) a blocking issue, and b) over two months old.

**Comment by tylrtrmbl** [comment]
@tonymillion: you added the [big warning](https://github.com/tonymillion/Reachability#warning-there-have-been-reports-of-apps-being-rejected-when-reachability-is-used-in-a-framework-the-only-solution-to-this-so-far-is-to-rename-the-class) in the README for this issue; but if you don't mind me saying, that's not really a solution that helps a CocoaPods user, which is where this issue was discovered. :wink: 

`TMReachability`?

**Comment by warpling** [comment]
I'm struggling a bit with this. I renamed Reachability's files and classes and successfully submitted to Apple but after a `pod install` I am getting linker errors with TMReachability. Is there a correct way to fix this via the pod or should I just ditch the pod in favor of my own fork or copying in the files manually?

**Comment by Cinamonas** [comment]
@warpling: as per @albertbori suggestion, that's why I'm doing at the moment:

```ruby
pod 'TMReachability', :git => 'https://github.com/albertbori/Reachability', :commit => 'e34782b386307e386348b481c02c176d58ba45e6'
```

**Comment by warpling** [comment]
Thanks @Cinamonas, I missed that! I ended up just pulling in the two files and renaming them.

**Comment by wasnotrice** [comment]
@albertbori thanks for sharing your fork. fixed this issue for me!

**Comment by dougdiego** [comment]
+1

**Comment by parallaxe** [comment]
+1

**Comment by DaveCole** [comment]
@Cinamonas solution somehow works for me when I run it on the phone, but fails when I attempt an App Store archive.  Anyone else run into this?

**Comment by warpling** [comment]
@DaveCole different target for release/debug perhaps? Can you run it on your phone in release mode? (You can change the mode in `Product > Edit Scheme`)

**Comment by dlo** [comment]
@Cinamonas Thanks so much for creating that fork! Works great. 👍
